### PR TITLE
Schedule field is working fix working.

### PR DESCRIPTION
**Pull request by rajeshcanaris**
Schedule field is working fix working.

### CODE REVIEW

File: app/Console/Commands/AutoReport.php [hunk 1]
```diff
@@ -48,7 +48,8 @@ public function __construct()
      */
     public function handle(Request $request)
     {
-
+        $this->info(Carbon::now()->format('l'));
+        // dd('done');
```
Comment (Shubham40Bears) [inline app/Console/Commands/AutoReport.php]: remove this commented out dd and $this->info

File: app/Console/Commands/AutoReport.php [hunk 2]
```diff
@@ -60,61 +61,69 @@ public function handle(Request $request)
...
-            $this->info(Carbon::now()->subDays(30)->startOfMonth()->endOfDay());
-            $tickets = $tickets->whereBetween('created_at',[Carbon::now()->subDays(30)->startOfMonth()->startOfDay(),Carbon::now()->subDays(30)->endOfMonth()->endOfDay()]);
-
-        }
-        else if(!empty($settings->frequency) && $settings->frequency === 'Weekly'){
-            $this->info(Carbon::now()->subDays(7)->startOfWeek()->endOfDay());
-            $tickets = $tickets->whereBetween('created_at',[Carbon::now()->subDays(7)->startOfWeek()->startOfDay(),Carbon::now()->subDays(7)->endOfWeek()->endOfDay()]);
+        if($settings->auto_data == 1){ // Start generating email report if auto reporting is turned on 
+            $emails=explode(',', $settings->to_email);
+            $this->info($settings->frequency);
```
Comment (Shubham40Bears) [inline app/Console/Commands/AutoReport.php]: remove all these infos

File: app/Console/Commands/AutoReport.php [hunk 3]
```diff
@@ -60,61 +61,69 @@ public function handle(Request $request)
...
+            $this->info($settings->schedule);
+        //    dd('done');
```
Comment (Shubham40Bears) [inline app/Console/Commands/AutoReport.php]: remove this dd

File: app/Console/Commands/AutoReport.php [hunk 4]
```diff
@@ -60,61 +61,69 @@ public function handle(Request $request)
...
+            file_put_contents('storage/slaReport/report.pdf', $output);
+            $data = array('name'=>"Example");
+
+            if($settings->frequency === 'Daily' && $settings->time === Carbon::now()->format('H:i:00') || $settings->frequency === 'Weekly'&& $settings->schedule === Carbon::now()->format('l') && $settings->time === Carbon::now()->format('H:i:00') || $settings->frequency === 'Monthly' && $settings->schedule === Carbon::now()->format('d') && $settings->time === Carbon::now()->format('H:i:00') ){
+                foreach($emails as $email){
+                    // $this->info($email);
+                $mail = Mail::send('mail', $data, function($message) use ($email, $settings){
+                $message->to($email, 'Canidesk User')->subject
+                    ($settings->frequency.' Report')->attach('storage/slaReport/report.pdf');
+                $message->from('rajeshcanaris@gmail.com','[ Canidesk Report ]');
```
Comment (Shubham40Bears) [inline app/Console/Commands/AutoReport.php]: change this from url to the MAIL_USERNAME variable from env file

**Review by Shubham40Bears — CHANGES_REQUESTED** [review]

File: app/Console/Commands/AutoReport.php [hunk 5]
```diff
@@ -57,64 +57,48 @@ public function handle(Request $request)
...
+            $dompdf = new Dompdf();
+            $options = new Options();
+            $options->set('defaultFont', 'Arial'); 
+            $dompdf->setOptions($options);
+            $html = view('sla.report-email', compact('tickets'));
+            $dompdf->load_html($html);
+            $dompdf->render();
+            $output = $dompdf->output();
+            file_put_contents('storage/slaReport/report.pdf', $output);
+            $data = array('name'=>"Example");
```
Comment (umar-ashrfy) [inline app/Console/Commands/AutoReport.php]: report.pdf should be

Report_{Weekly}_{start_date}_{end_date}.pdf

Comment (umar-ashrfy) [inline app/Console/Commands/AutoReport.php]: Example => Canidesk

**Review by umar-ashrfy — CHANGES_REQUESTED** [review]

**Review by umar-ashrfy — APPROVED** [review]

**Pull request merged**